src/Instrumentation/MySqli/src/MySqliInstrumentation.php

```diff
@@ -440,11 +448,52 @@ private static function constructPostHook(int $paramsOffset, CachedInstrumentati
     }
 
     /** @param non-empty-string $spanName */
-    private static function queryPreHook(string $spanName, CachedInstrumentation $instrumentation, MySqliTracker $tracker, $obj, array $params, ?string $class, string $function, ?string $filename, ?int $lineno): void
+    private static function queryPreHook(string $spanName, ?TextMapPropagatorInterface $contextPropagator, CachedInstrumentation $instrumentation, MySqliTracker $tracker, $obj, array $params, ?string $class, string $function, ?string $filename, ?int $lineno): array
     {
         $span = self::startSpan($spanName, $instrumentation, $class, $function, $filename, $lineno, []);
         $mysqli = $obj ? $obj : $params[0];
+        $query = $obj ? $params[0] : $params[1];
+        $query = mb_convert_encoding($query ?? self::UNDEFINED, 'UTF-8');
+        if (!is_string($query)) {
+            $query = self::UNDEFINED;
+        }
+        $span->setAttributes([
+            TraceAttributes::DB_QUERY_TEXT => $query,
+            TraceAttributes::DB_OPERATION_NAME => self::extractQueryCommand($query),
+        ]);
+
         self::addTransactionLink($tracker, $span, $mysqli);
+
+        if (class_exists('OpenTelemetry\Contrib\SqlCommenter\SqlCommenter') && $query !== self::UNDEFINED) {
+            $comments = [];
+            if ($contextPropagator !== null) {
+                $contextPropagator->inject($comments);
+            } else {
+                Globals::propagator()->inject($comments);
+            }
+            if (class_exists('OpenTelemetry\Contrib\SqlCommenter\SqlCommenter')) {
+                $query = \OpenTelemetry\Contrib\SqlCommenter\SqlCommenter::inject($query, $comments);
+            }
+            if (class_exists('OpenTelemetry\Contrib\ContextPropagator\ContextPropagation') && \OpenTelemetry\Contrib\ContextPropagator\ContextPropagation::isAttributeEnabled()) {
+                $span->setAttributes([
+                    TraceAttributes::DB_QUERY_TEXT => (string) $query,
+                    TraceAttributes::DB_OPERATION_NAME => self::extractQueryCommand($query),
```

Comment:
Why does the `DB_OPERATION_NAME` need to be set again?

Reply:
DB_QUERY_TEXT and DB_OPERATION_NAME are set in pair. I would like to make sure the DB_OPERATION_NAME is extracted from the updated DB_QUERY_TEXT.

Reply:
According to the [spec](https://opentelemetry.io/docs/specs/semconv/database/database-spans/), `db.operation.name` is supposed to be the low cardinality operation such as `SELECT` extracted from query text. Does the `extractQueryCommand` actually include comments!? if so it really shouldn't. On assumption it doesn't, why do this possibly expensive work twice?